src/framework/theme/components/popover/popover.component.ts

```diff
@@ -61,6 +70,24 @@ export class NbPopoverComponent {
   @HostBinding('style.left.px')
   positionLeft: number;
 
+  /**
+   * If content type is TemplateRef we're passing context as template outlet param.
+   * But if we have custom component content we're just assigning passed context to the component instance.
+   * */
+  @ViewChild(NgComponentOutlet)
+  set componentOutlet(el) {
+    if (this.isComponent) {
+      Object.assign(el._componentRef.instance, this.context);
```

Comment:
is this the best way to access the component reference?

Reply:
Yep, because we don't have component type and can query only by *NgComponentOutlet*

Reply:
at least we can do `@ViewChild(NgComponentOutlet, {read: ViewContainerRef})`

Reply:
Ok, I'll check it